pages/dashboard/index.js

```diff
@@ -173,39 +184,42 @@ const redirectToHome = {
   },
 };
 
-export const getServerSideProps = async ({ req, locale }) => {
-  const apolloClient = initializeApollo();
-  await apolloClient.query({
-    query: GET_USERS,
-  });
-
-  const session = await getSession({ req });
+export const getServerSideProps = async (context) => {
+  const session = await getSession(context);
   if (!session) {
+    console.error('Log in first!');
     return redirectToHome;
   }
 
-  await dbConnect();
+  const { req, locale } = context;
+  const { name, email } = session.user;
+
+  const apolloClient = initializeApollo();
 
   let newSession = {};
-  if (session) {
-    try {
-      const user = await User.findOne({
-        email: session.user.email,
-      });
-      if (!user) {
-        return redirectToHome;
-      }
-      newSession = { ...session, user };
-      console.debug('Session:', newSession);
-    } catch (err) {
-      console.error(err);
-      return redirectToHome;
-    }
+  try {
+    const {
+      data: { users },
+    } = await apolloClient.query({
+      query: GET_USERS,
+      variables: { usersName: name, usersEmail: email },
+    });
+
+    const userData = users[0];
+
+    // Add friend and id fields to user object
+    session.user = { ...session.user, ...userData };
+    newSession = { ...session };
+    console.debug('newSession', newSession);
+  } catch (error) {
+    console.debug('Log in first', error);
```

Comment:
Not a big deal but in general, when prefixing a messaging, add a colon to separate the prefix from the information. Ex. `console.debug('newSession:', newSession`. Doesn't matter too much when the info is a object, but when it's a string, then a colon adds clarity. TBH I just add colons for all console debugs to keep things consistent.